### DISCUSSION TITLE
value of getSession is null in getServerSideProps and getInitialProps

### DISCUSSION THREAD

**Discussion by Industrial** [nextauthjs/next-auth#1693] [Help]
**Describe the bug**
When I clone the example project from https://github.com/nextauthjs/next-auth-example and start the server and go to the `/server` route, the session variable is `null`.

**Steps to reproduce**
```
$ git clone https://github.com/nextauthjs/next-auth-example
$ cd next-auth-example
$ yarn
$ yarn dev
```

Now go to https://github.com/nextauthjs/next-auth-example/blob/main/pages/_app.js#L7 and add a `console.log(pageProps.session)`.

Now open the browser and visit http://localhost:3000/server

**Expected behavior**
A session object.

**Screenshots or error logs**

Server:
```
[next-auth][warn][nextauth_url]
https://next-auth.js.org/warnings#nextauth_url NEXTAUTH_URL environment variable not set
null
```

Client:
```
null _app.js:7:10
[Fast Refresh] done hot-dev-client.js:182:2
[Fast Refresh] rebuilding hot-dev-client.js:196:6
```

**Additional context**
Currently having problems logging in so I only checked the logged case in this app but it does the same (`null`) in my own app while logged in.

I also tried replacing the getServerSideProps with getInitialProps (because of https://nextjs.org/docs/advanced-features/custom-app#caveats)

**Feedback**
*Documentation refers to searching through [online documentation](https://next-auth.js.org), code comments and issue history. The example project refers to [next-auth-example](https://github.com/iaincollins/next-auth-example).*

* [x] Found the documentation helpful
* [ ] Found documentation but was incomplete
* [ ] Could not find relevant documentation
* [ ] Found the example project helpful
* [ ] Did not find the example project helpful

**Comment by dcolonv** [comment]
You can create a `.env` file on the root of the project, add a line `NEXTAUTH_URL=http://localhost:3000`, and restart the server. NEXTAUTH_URL environment variable is needed to start the api

**Comment by balazsorban44** [comment]
I am not entirely sure why you expected a session object. You must log in with one of the providers to see it. I just did what you said to reproduce, but I got the user `session` without a problem.
